core/provisioning-java/src/test/java/org/apache/syncope/core/provisioning/java/pushpull/LDAPMembershipPullActionsTest.java

```diff
@@ -0,0 +1,177 @@
+import org.apache.syncope.core.persistence.api.dao.AnyTypeDAO;
+import org.apache.syncope.core.persistence.api.dao.GroupDAO;
+import org.apache.syncope.core.persistence.api.entity.AnyType;
+import org.apache.syncope.core.persistence.api.entity.ConnInstance;
+import org.apache.syncope.core.persistence.api.entity.resource.ExternalResource;
+import org.apache.syncope.core.persistence.api.entity.task.ProvisioningTask;
+import org.apache.syncope.core.persistence.api.entity.user.UMembership;
+import org.apache.syncope.core.persistence.api.entity.user.User;
+import org.apache.syncope.core.persistence.jpa.entity.JPAAnyType;
+import org.apache.syncope.core.persistence.jpa.entity.resource.JPAProvision;
+import org.apache.syncope.core.persistence.jpa.entity.user.JPAUMembership;
+import org.apache.syncope.core.persistence.jpa.entity.user.JPAUser;
+import org.apache.syncope.core.provisioning.api.Connector;
+import org.apache.syncope.core.provisioning.api.pushpull.ProvisioningProfile;
+import org.apache.syncope.core.provisioning.api.pushpull.ProvisioningReport;
+import org.apache.syncope.core.provisioning.java.AbstractTest;
+import org.identityconnectors.framework.common.objects.Attribute;
+import org.identityconnectors.framework.common.objects.ConnectorObject;
+import org.identityconnectors.framework.common.objects.SyncDelta;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.quartz.JobExecutionException;
+import org.springframework.test.util.ReflectionTestUtils;
+
+public class LDAPMembershipPullActionsTest extends AbstractTest {
+
+    @Mock
+    private AnyTypeDAO anyTypeDAO;
+
+    @Mock
+    private GroupDAO groupDAO;
+
+    @Mock
+    private InboundMatcher inboundMatcher;
+
+    @InjectMocks
+    private LDAPMembershipPullActions ldapMembershipPullActions;
+
+    @Mock
+    private SyncDelta syncDelta;
+
+    @Mock
+    private ProvisioningProfile<?, ?> profile;
+
+    @Mock
+    private ProvisioningReport result;
+
+    private EntityTO entity;
+
+    private AnyPatch anyPatch;
+
+    private Map<String, Set<String>> membershipsBefore;
+
+    private User user;
+
+    @BeforeEach
+    public void before() {
+        List<UMembership> uMembList = new ArrayList<>();
+        UMembership uMembership = new JPAUMembership();
+        user = new JPAUser();
+        uMembership.setLeftEnd(user);
+        ReflectionTestUtils.setField(user, "id", UUID.randomUUID().toString());
+        uMembList.add(uMembership);
+
+        anyPatch = new UserPatch();
+        membershipsBefore = new HashMap<>();
+        ReflectionTestUtils.setField(ldapMembershipPullActions, "membershipsBefore", membershipsBefore);
+
+        lenient().when(groupDAO.findUMemberships(groupDAO.find(anyString()))).thenReturn(uMembList);
+    }
+
+    @Test
+    public void testBeforeUpdateWithGroupTOAndEmptyMemberships() throws JobExecutionException {
```

Comment:
please rename all methods by removing the `test` prefix: not needed.

Reply:
Methods have been renamed